app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -348,17 +355,103 @@ private void setRoots() {
     }
 
     private int initializeAppResources() {
-        //There should be exactly one of these for now
+        // Before we try to initialize a new app, check if any existing apps were left in a
+        // partially deleted state, and finish uninstalling them if so
         for (ApplicationRecord record : getGlobalStorage(ApplicationRecord.class)) {
-            if (record.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
-                //We have an app record ready to go
+            if (record.getStatus() == ApplicationRecord.STATUS_DELETE_REQUESTED) {
+                record.uninstall(this);
+            }
+        }
+        
+        // There may now be multiple app records in storage, because of multiple apps support.
+        // Since this method is called from onCreate of CommCareApplication, current behavior is
+        // just to pick the first app in the list to initialize
+        for(ApplicationRecord record : getGlobalStorage(ApplicationRecord.class)) {
+            if(record.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
                 return initializeAppResources(new CommCareApp(record));

```

Comment:
I'm a little irked by this behavior, though I don't know what a good solution would be:
When CommCare starts up, this will register all the resources in the global table with the platform object. I assume this is expensive enough that we want to avoid superfluous initializations. It seems that with multiple apps, the behavior will be to initialize the first app that is installed, even if it is the one we aren't going to use, and then at some point later register the one that the user actually wants to use. 

Is my reasoning on the right track?

Could we delay this initialization until the point where we know what app the user actually wants to use?